gmso/tests/test_topology.py

```diff
@@ -418,4 +418,23 @@ def test_topology_atom_type_changes(self):
         assert top.sites[10].atom_type.name == 'atom_type_changed'
         assert top.is_typed()
 
+    def test_get_associations_atom_types(self):
+        top = Topology()
+        atom_type = AtomType(name='test_atomtype')
+        for i in range(10):
+            top.add_site(Site(name=f'site{i+1}', atom_type=atom_type), update_types=True)
+        assert len(top.get_associations(atom_type)) == 10
+
+    def test_get_association_atom_types_after_changes(self, typed_ar_system):
+        typed_ar_system.atom_types[0].name = 'Typed Ar System'
+        assert typed_ar_system.get_associations(typed_ar_system.atom_types[0]) == set(typed_ar_system.sites)
+
+    def test_get_association_bond_types(self, typed_water_system):
```

Comment:
not sure if this fits here, but what if we had a system with bonds (water would be an example: H-O-H)

In this case, bondtype should have a set that contains all 3 sites. If we then removed a bond  between one of the H and O (H     O-H)

We expect that the `len(get_associations(broken_water)` is now 2.

Would be a good test to have. and this same example should work for angletype too.